### PR TITLE
Use full path when running sanity check

### PR DESCRIPTION
**Pull request by jovial**
```
Run .github/workflows/collection-sanity.sh
/home/runner/work/_temp/54a8cac7-3453-49be-a426-1e966f9798bc.sh: line 1: .github/workflows/collection-sanity.sh: No such file or directory
Error: Process completed with exit code 127.
```

See: https://github.com/stackhpc/ansible-collection-linux/actions/runs/5655376499/job/15320367246?pr=10#step:5:12

### CODE REVIEW

File: .github/workflows/lint-collection.yml [hunk 1]
```diff
@@ -41,6 +41,8 @@ jobs:
     steps:
       # Checks-out the repository under $GITHUB_WORKSPACE, so it's accessible to the job
       - uses: actions/checkout@v3
+        with:
+          path: stackhpc-github
```
Comment (markgoddard) [inline .github/workflows/lint-collection.yml:45]: This will be the collection repo, not .github. You might need another checkout step

Comment (jovial) [reply .github/workflows/lint-collection.yml:45]: Good point. Thing is I'm not sure there is a context variable containing the git_ref for this reusable workflow i.e need to get main from: 

```
  lint:
    uses: stackhpc/.github/.github/workflows/lint-collection.yml@main
```
Wonder whether its easier just to include it inline in the workflow?

Comment (markgoddard) [reply .github/workflows/lint-collection.yml:45]: heh, this should have been easy. Might be simplest to just checkout main of .github

Comment (jovial) [reply .github/workflows/lint-collection.yml:49]: Nightmare in 2.12:

```
This version of ansible-test cannot be executed with Python version 3.11.4. Supported Python versions are: 3.8, 3.9, 3.10
```

https://github.com/stackhpc/ansible-collection-linux/actions/runs/5657449401/job/15326522093?pr=11#step:6:25

**Review by markgoddard — APPROVED** [review]
Thank you for getting this over the line.

**Pull request merged**